retro/reco.py

```diff
@@ -1686,15 +1694,15 @@ def func(x):
         fit_meta = OrderedDict(
             [
                 ("fit_status", np.int8(fit_status)),
-                ("iterations", np.uint32(iter_num)),
+                ("iterations", np.int32(iter_num)),
                 ("stopping_flag", np.int8(stopping_flag)),
                 ("llh_std", np.float32(fit['meta']['fstd'])),
-                ("no_improvement_counter", np.uint32(fit['meta']['no_improvement_counter'])),
+                ("no_improvement_counter", np.int32(fit['meta']['no_improvement_counter'])),
```

Comment:
why the changes in types here? (it seems fine to make the change, but I'm just curious if there is a negative "special" value or something somewhere that necessitates int vs. uint)

Reply:
there is the value `-1` is it is not initialized, and i saw it then creates overflow. Since with the new i3 format we will anyway convert to signed int, I thought i do it here directly instead of int->uint->int

Reply:
Very well, glad you noticed that.